stats/bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java

```diff
@@ -223,9 +224,11 @@ private void registerMetrics(Collector collector) {
         AtomicLong tmpDirectMemoryUsage = null;
 
         try {
-            Field field = PlatformDependent.class.getDeclaredField("DIRECT_MEMORY_COUNTER");
-            field.setAccessible(true);
-            tmpDirectMemoryUsage = (AtomicLong) field.get(null);
+            Class<?> name = Class.forName("java.nio.Bits");
+            String fieldName = SystemUtils.IS_JAVA_1_8 ? "reservedMemory" : "RESERVED_MEMORY";
+            Field reservedMemory = name.getDeclaredField(fieldName);
+            reservedMemory.setAccessible(true);
+            tmpDirectMemoryUsage = (AtomicLong) reservedMemory.get(null);
```

Comment:
fyi, you can use org.apache.commons.lang.reflect.FieldUtils [FieldUtils.readDeclaredField() ](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/reflect/FieldUtils.html#readField-java.lang.reflect.Field-java.lang.Object-boolean-)with forceAccess